fix: Install bazaar for apps using go.mod

shackra opened the pull request:
- fixes #344

`make testpack` fails leaving me in a shell inside Docker with an error saying "/buildpack/bin/test: line 46: go: command not found" so not sure how to test my changes.

shackra committed:
fix: Install bazaar for apps using go.mod

- fixes #344

freeformz commented:
Try `make test` as `make testpack` is for testing the testpack and I'll have to look into that separately.

freeformz commented:
PS: Please add a test fixture that includes a dependency that needs to be fetched over bazaar.

Commit:
Add modules+bzr test / add bzr file checking

Add a test that includes a dependency that is using bzr so we have
some confidence going forward.

Add the source and sha for bzr 2.7.0 to files.json and use ensureFile
to make sure it's validated at download time.

Commit:
Update the changelog

freeformz commented:
@shackra I rebased and pushed commits that add tests, ensure the downloaded source file matches the recorded sha, etc.

jmorrell commented on bin/compile:
```diff
@@ -480,6 +501,9 @@ setupProcfile() {
 case "${TOOL}" in
     gomodules)
+        ensureBZR "${BazaarVersion}"
+        ensureHG "${MercurialVersion}"
```
Were we not installing mercurial before?

elzorrorebelde replied:
hi!

Maybe that was the case but I don't remember

shackra commented:
Thanks @freeformz I was too busy to give it a try and add the tests myself
